### PR TITLE
[CIS-1107] Fix thread avatar view not displaying latest reply author avatar

### PR DESCRIPTION
**Pull request by nuno-vieira**
## Description of the pull request
Fixes the avatar of the latest reply not showing the image correctly on the first loading of the message list. The avatar was being fetched by the `latestReplies` property which is only available after loading all the replies. The proper way to do this is to get the author of the last reply through the `threadParticipants` which are ordered by activity. This introduces a breaking change since currently, the `threadPariticipants` is a Set instead of an Array.

This PR also fixes the CI.

## Breaking Change
- `ChatMessage.threadParticipants: Set<ChatUser>` -> `ChatMessage.threadParticipants: [ChatUser]`

### CODE REVIEW

**Comment by tbarbugli** [comment]
@nuno-vieira https://github.com/GetStream/stream-chat-swift/pull/1398/checks?check_run_id=3404905004#step:7:1326 CI is not compiling

File: Sources/StreamChat/Database/DTOs/MessageDTO.swift [hunk 1]
```diff
@@ -26,7 +26,7 @@ class MessageDTO: NSManagedObject {
     
     @NSManaged var user: UserDTO
     @NSManaged var mentionedUsers: Set<UserDTO>
-    @NSManaged var threadParticipants: Set<UserDTO>
+    @NSManaged var threadParticipants: NSOrderedSet
```
Comment (tbarbugli) [inline Sources/StreamChat/Database/DTOs/MessageDTO.swift:29]: why NSOrderedSet and not an Array?

**Review by evsaev — APPROVED** [review]
LGTM 👍

File: Sources/StreamChat/Database/StreamChatModel.xcdatamodeld/StreamChatModel.xcdatamodel/contents [hunk 2]
```diff
@@ -185,7 +185,7 @@
         <relationship name="quotedMessage" optional="YES" maxCount="1" deletionRule="Nullify" destinationEntity="MessageDTO" inverseName="quotedBy" inverseEntity="MessageDTO"/>
         <relationship name="reactions" optional="YES" toMany="YES" deletionRule="Nullify" destinationEntity="MessageReactionDTO" inverseName="message" inverseEntity="MessageReactionDTO"/>
         <relationship name="replies" optional="YES" toMany="YES" deletionRule="Nullify" destinationEntity="MessageDTO" inverseName="parentMessage" inverseEntity="MessageDTO"/>
-        <relationship name="threadParticipants" optional="YES" toMany="YES" deletionRule="Nullify" destinationEntity="UserDTO" inverseName="participatedThreads" inverseEntity="UserDTO"/>
+        <relationship name="threadParticipants" optional="YES" toMany="YES" deletionRule="Nullify" ordered="YES" destinationEntity="UserDTO" inverseName="participatedThreads" inverseEntity="UserDTO"/>
```
Comment (evsaev) [inline Sources/StreamChat/Database/StreamChatModel.xcdatamodeld/StreamChatModel.xcdatamodel/contents:188]: nit: would be nice to have a test on `ChatMessage` that verifies the thread participants are exposed in the same order they were ordered in backend response

Comment (nuno-vieira) [reply Sources/StreamChat/Database/StreamChatModel.xcdatamodeld/StreamChatModel.xcdatamodel/contents:188]: I've added more thread participants to the dummy payloads before it was just one item. The current tests already check that the array is equal 👍

**Review by evsaev — DISMISSED** [review]

**Review by evsaev — APPROVED** [review]
🎉

**Pull request merged**